react/features/local-recording/recording/flac/flacEncodeWorker.js

```diff
@@ -0,0 +1,416 @@
+import {
+    MAIN_THREAD_FINISH,
+    MAIN_THREAD_INIT,
+    MAIN_THREAD_NEW_DATA_ARRIVED,
+    WORKER_BLOB_READY,
+    WORKER_LIBFLAC_READY
+} from './messageTypes';
+
+/**
+ * WebWorker that does FLAC encoding using libflac.js
+ */
+
+/* eslint-disable */
+importScripts('/libs/libflac3-1.3.2.min.js');
+/* eslint-enable */
+
+// There is a number of API calls to libflac.js, which does not conform
+// to the camalCase naming convention, but we cannot change it.
+// So we disable the ESLint rule `new-cap` in this file.
+/* eslint-disable new-cap */
+
+// Flow will complain about the number keys in `FLAC_ERRORS,
+// ESLint will complain about the `declare` statement.
+// As the current workaround, add an exception for eslint.
+/* eslint-disable flowtype/no-types-missing-file-annotation*/
+declare var Flac: Object;
+
+const FLAC_ERRORS = {
+    // The encoder is in the normal OK state and
+    // samples can be processed.
+    0: 'FLAC__STREAM_ENCODER_OK',
+
+    // The encoder is in the
+    // uninitialized state one of the FLAC__stream_encoder_init_*() functions
+    // must be called before samples can be processed.
+    1: 'FLAC__STREAM_ENCODER_UNINITIALIZED',
+
+    // An error occurred in the underlying Ogg layer.
+    2: 'FLAC__STREAM_ENCODER_OGG_ERROR',
+
+    // An error occurred in the
+    // underlying verify stream decoder; check
+    // FLAC__stream_encoder_get_verify_decoder_state().
+    3: 'FLAC__STREAM_ENCODER_VERIFY_DECODER_ERROR',
+
+    // The verify decoder detected a mismatch between the
+    // original audio signal and the decoded audio signal.
+
+    4: 'FLAC__STREAM_ENCODER_VERIFY_MISMATCH_IN_AUDIO_DATA',
+
+    // One of the callbacks returned
+    // a fatal error.
```

Comment:
So someone might point out to you that you should try to fill out the 80 character line limit as much as you can before line breaking. This applies to all comments.

Reply:
Fixed.